controls/cf_serverd.cf

```diff
@@ -45,13 +45,29 @@ body server control
       cfruncommand => "$(sys.cf_agent) -I -D cfruncommand -f \"$(sys.update_policy_path)\"  &
                        $(sys.cf_agent) -I -D cfruncommand";
 
-    !windows::
-      # last single quote in cfruncommand is left open, so that
-      # arguments (like -K and --remote-bundles) are properly appended.
-      cfruncommand => "$(def.cf_runagent_shell) -c \'
+    !windows.!cfengine_3_7::
+
+      # In 3.10 the quotation is properly closed when EOF is reached. It is left
+      # open so that arguments (like -K and --remote-bundles) can be appended.
+      # 3.10.x does not automatically append -I -Dcfruncommand
+
+        cfruncommand => "$(def.cf_runagent_shell) -c \'
                            $(sys.cf_agent) -I -D cfruncommand -f $(sys.update_policy_path)  ;
                            $(sys.cf_agent) -I -D cfruncommand";
 
+    !windows.cfengine_3_7::
+
+      # 3.7.x does not properly close open quotes when EOF is reached preventing
+      # policy execution, so we must provide a trailing quote for cfruncommand.
+
+      # 3.7.x Automatically appends -I -Dcfruncommand to the cfruncommand. This
+      # results in a benign error about not being allowed to specify inputs for
+      # remote execution.
```

Comment:
I don't understand why you say it's a benign error.  This won't run the update.cf file at all on CFEngine 3.7.3.

Reply:
I thought the error was related to the automatic appending of  `-I -Dcfruncommand`. But it was indeed from the update policy activation.

Reply:
Great!  I would just recommend removing the inaccurate comment, otherwise it looks good to me.  I'm not convinced that the `cf_runagent_initiated` class is *necessary*, but it's not harmful.  :)

Reply:
That's right.  Thanks, Nick.